### PR TITLE
[IMPROVE][ENTERPRISE] UI/UX enhancements in Omnichannel Units page

### PR DESCRIPTION
**Pull request by rafaelblink**
## Proposed changes

Some changes to follow Rocket.chat's design system:

List page:
* "New" button change
* Grid columns width redefinition
* Implementing search filter on REST API

New/edit page:
* Required fields refactor
* Buttons following design system

## Screenshots

**Before**
image
image

**After**

image
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ee/app/livechat-enterprise/server/api/lib/units.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 import LivechatUnitMonitors from '../../../../models/server/models/LivechatUnitMonitors';
 
-export async function findUnits({ userId, pagination: { offset, count, sort } }) {
+export async function findUnits({ userId, text, pagination: { offset, count, sort } }) {
 	if (!await hasPermissionAsync(userId, 'manage-livechat-units')) {
 		throw new Error('error-not-authorized');
 	}
-	const cursor = LivechatUnit.find({}, {
+
+	const query = { ...text && { $or: [{ name: new RegExp(s.escapeRegExp(text), 'i') }, { _id: new RegExp(s.escapeRegExp(text), 'i') }] } };
```
Comment (renatobecker) [inline ee/app/livechat-enterprise/server/api/lib/units.js:12]: Why are you finding units by _id using the text parameter?
IMO filtering  `name` field only is enough.

File: ee/client/omnichannel/units/UnitsRoute.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ function UnitsRoute() {
 
 	const header = useMemo(() => [
 		<GenericTable.HeaderCell key={'name'} direction={sort[1]} active={sort[0] === 'name'} onClick={onHeaderClick} sort='name' w='x120'>{t('Name')}</GenericTable.HeaderCell>,
-		<GenericTable.HeaderCell key={'visibility'} direction={sort[1]} active={sort[0] === 'visibility'} onClick={onHeaderClick} sort='visibility' w='x200'>{t('Visibility')}</GenericTable.HeaderCell>,
+		<GenericTable.HeaderCell key={'visibility'} direction={sort[1]} active={sort[0] === 'visibility'} onClick={onHeaderClick} sort='visibility' w='x60'>{t('Visibility')}</GenericTable.HeaderCell>,
```
Comment (renatobecker) [inline ee/client/omnichannel/units/UnitsRoute.js:104]: The remove button isn't aligned properly anymore.

Screen Shot 2020-11-18 at 10 12 39

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: ee/client/omnichannel/units/UnitsRoute.js (hunk 2 above)
Comment (renatobecker) [reply ee/client/omnichannel/units/UnitsRoute.js]: @rafaelblink I'm still facing the same issue. The changes you submitted didn't solve the issue.

Screen Shot 2020-11-18 at 13 56 36

Comment (rafaelblink) [reply ee/client/omnichannel/units/UnitsRoute.js]: This is because the width of columns needs to be auto. I'm pushing a possible fix.

**Review by renatobecker — APPROVED** [review]

**Pull request merged**